internal/responses/responses.go

```diff
@@ -29,7 +29,8 @@ func RespondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
 	_, _ = w.Write(response)
 }
 
-// RespondWithError writes a JSON response with an error message and status code to the HTTP ResponseWriter.
+// RespondWithError writes a response with an error message and status code to the HTTP ResponseWriter.
 func RespondWithError(w http.ResponseWriter, code int, message string) {
-	RespondWithJSON(w, code, map[string]string{"error": message})
+	w.WriteHeader(code)
+	_, _ = w.Write([]byte(message))
```

Comment:
Who or what is setting the content type in this response? The documentation isn't super clear on this.

Reply:
This was a rabbit-hole :) I'm pretty sure it is written here (same version of Golang as in the Dockerfile):
https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/net/http/server.go;l=1386

But it scares me a bit that we are letting Golang detect the content type so I will update the code to be more explicit :)

Reply:
Ah, yes, that's where the autodetection kicks in. Being explicit about this SGTM.